### PR TITLE
Implement Waitlist api endpoint

### PR DESCRIPTION
**Pull request by gurumarv**
This pull request adds the following:
- `EmailProducerService`: Service for sending email messages.
- `EmailConsumerService`: Service for receiving and processing email messages.
- `RabbitMQConfig`: Configuration for RabbitMQ.
- `EmailMessage`: Entity class for email messages.

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/waitlist/controller/WaitlistController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private WaitlistService waitlistService;
+
+    @PostMapping
+    public ResponseEntity<?> createWaitlist(@Valid @RequestBody Waitlist waitlist){
+        waitlistService.saveWaitlist(waitlist);
+        Map<String, String> response = new HashMap<>();
+        response.put("message", "You are all signed up!");
+        return new ResponseEntity<>(response, HttpStatus.CREATED);
+    }
+
```
Comment (Am0du) [inline src/main/java/hng_java_boilerplate/waitlist/controller/WaitlistController.java]: Please remove the empty space

File: src/main/resources/db/migration/V1__init_database.sql [hunk 2]
```diff
@@ -39,4 +40,11 @@ CREATE TABLE user_organisation (
     PRIMARY KEY (user_id, organisation_id),
     FOREIGN KEY (user_id) REFERENCES users(id),
     FOREIGN KEY (organisation_id) REFERENCES organisations(id)
+);
+
+-- V1__Create_waitlist_table.sql
```
Comment (Am0du) [inline src/main/resources/db/migration/V1__init_database.sql:47]: This should be in a new Flyway migration script with a unique version number.

**Review by Am0du — COMMENTED** [review]
I also don't see unit test

File: src/main/java/hng_java_boilerplate/waitlist/controller/WaitlistController.java (hunk 1 above)
Comment (gurumarv) [reply src/main/java/hng_java_boilerplate/waitlist/controller/WaitlistController.java]: I have added the unit test and it passed

File: src/main/resources/db/migration/V1__init_database.sql (hunk 2 above)
Comment (gurumarv) [reply src/main/resources/db/migration/V1__init_database.sql:47]: That is not my code oh, boss

File: src/main/java/hng_java_boilerplate/waitlist/filter/RateLimitFilter.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+public class RateLimitFilter implements Filter {
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/waitlist/filter/RateLimitFilter.java:14]: Make Rate Limiter independent of the waitlist so any service can call it

File: src/main/java/hng_java_boilerplate/waitlist/service/RateLimitService.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package hng_java_boilerplate.waitlist.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Service
+public class RateLimitService {
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/waitlist/service/RateLimitService.java:10]: Make Rate Limiter independent of the waitlist so any service can call it

File: src/main/java/hng_java_boilerplate/waitlist/filter/RateLimitFilter.java
Comment (Olaoluwa22) [inline src/main/java/hng_java_boilerplate/waitlist/filter/RateLimitFilter.java:1]: Wouldn't it be good to make this a util, so everyone can use?

**Pull request closed without merging**